src/ApiClient/ServerClient.ts

```diff
@@ -337,187 +337,30 @@ export default new (class ServerClient extends ApiClient<IEvents> {
 
     //Utility function that returns a promise which resolves with the token whenever theres valid credentials(could be immediatly)
     public wait4Token() {
-        return new Promise<TokenResponse>(resolve => {
-            if (CredentialsProvider.isTokenValid()) {
-                resolve(CredentialsProvider.token!);
-                return;
-            }
-            this.on("tokenAvailable", token => {
-                resolve(token);
-            });
-        });
+        return AuthController.getToken();
```

Comment:
I'd rather nuke the function if AuthController.getToken is what we're going with